env0/resource_environment.go

```diff
@@ -248,6 +248,26 @@ func setEnvironmentConfigurationSchema(d *schema.ResourceData, configurationVari
 func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	apiClient := meta.(client.ApiClientInterface)
 
+	if _, ok := d.GetOk("template.0"); ok {
+		// Templateless environment creation use-case.
+		createTemplatePayload, diagErr := templateCreatePayloadFromParameters("template.0", d)
+		if diagErr != nil {
+			return diagErr
+		}
+
+		if revision, ok := d.GetOk("revision"); ok {
+			createTemplatePayload.Revision = revision.(string)
+		}
+
+		template, err := apiClient.TemplateCreate(createTemplatePayload)
+		if err != nil {
+			return diag.Errorf("could not create template: %v", err)
+		}
+
+		d.Set("template.0.id", template.Id)
```

Comment:
This is important. Set the id after creating the template.

Reply:
@ItamarMalka - yes you're absolutely right!
The acceptance test hasn't been implemented yet (it's just a "planOnly" at the moment.
Because it's "plan only" the field won't be set.
The changes are done iteratively. Once I implement the read function, I will be able to implement the test.

```
t.Run("Success in create", func(t *testing.T) {
		testCase := resource.TestCase{
			Steps: []resource.TestStep{
				{
					Config: createEnvironmentResourceConfig(environment, template),
					Check: resource.ComposeAggregateTestCheckFunc(
						resource.TestCheckResourceAttr(accessor, "id", environment.Id),
						resource.TestCheckResourceAttr(accessor, "name", environment.Name),
						resource.TestCheckResourceAttr(accessor, "project_id", environment.ProjectId),
						resource.TestCheckNoResourceAttr(accessor, "template_id"),
						resource.TestCheckResourceAttr(accessor, "workspace", environment.WorkspaceName),
						resource.TestCheckResourceAttr(accessor, "terragrunt_working_directory", environment.TerragruntWorkingDirectory),
						resource.TestCheckResourceAttr(accessor, "vcs_commands_alias", environment.VcsCommandsAlias),
						resource.TestCheckResourceAttr(accessor, "revision", template.Revision),
						resource.TestCheckResourceAttr(accessor, "template.0.repository", template.Repository),
						resource.TestCheckResourceAttr(accessor, "template.0.terraform_version", template.TerraformVersion),
						resource.TestCheckResourceAttr(accessor, "template.0.type", template.Type),
						resource.TestCheckResourceAttr(accessor, "template.0.path", template.Path),
						resource.TestCheckResourceAttr(accessor, "template.0.retries_on_deploy", strconv.Itoa(template.Retry.OnDeploy.Times)),
						resource.TestCheckResourceAttr(accessor, "template.0.retry_on_deploy_only_when_matches_regex", template.Retry.OnDeploy.ErrorRegex),
						resource.TestCheckResourceAttr(accessor, "template.0.retries_on_destroy", strconv.Itoa(template.Retry.OnDestroy.Times)),
						resource.TestCheckResourceAttr(accessor, "template.0.retry_on_destroy_only_when_matches_regex", template.Retry.OnDestroy.ErrorRegex),
					),
					PlanOnly:           true,
					ExpectNonEmptyPlan: true,
				},
			},
		}

		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {

		})
	})
```